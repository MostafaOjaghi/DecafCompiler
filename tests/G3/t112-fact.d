int fact(int x) {
	if (x == 0)
		return 1;
	return x * fact(x - 1);
}

void main() {
	int n;
	int i;
	n = ReadInteger();
	for (i = 0; i < n; i = i + 1) {
		int a;
		a = ReadInteger();
		Print(fact(a)); 
	}
}
