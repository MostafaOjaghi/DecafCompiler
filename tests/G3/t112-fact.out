120
40320
2
3628800
